packages/appium/lib/config.js

```diff
@@ -50,11 +52,59 @@ const getFullGitPath = _.memoize(async function getFullGitPath() {
   }
 });
 
+/**
+ * Prints server debug into into the console.
+ *
+ * @param {{
+ * driverConfig: import('./extension/driver-config').DriverConfig,
+ * pluginConfig: import('./extension/plugin-config').PluginConfig,
+ * appiumHome: string
+ * }} info
+ * @returns {Promise<void>}
+ */
+export async function showDebugInfo({driverConfig, pluginConfig, appiumHome}) {
+  const getNpmVersion = async () => {
+    const {stdout} = await npm.exec('--version', [], {cwd: process.cwd()});
+    return _.trim(stdout);
+  };
+  const findNpmLocation = async () => await fs.which(system.isWindows() ? 'npm.cmd' : 'npm');
+  const [npmVersion, npmLocation] = await B.all(
+    [getNpmVersion, findNpmLocation].map((f) => getSafeResult(f, 'unknown'))
+  );
+  const debugInfo = {
+    os: {
+      platform: os.platform(),
+      release: os.release(),
+      arch: os.arch(),
+      homedir: os.homedir(),
+      username: os.userInfo().username,
+    },
+    node: {
+      version: process.version,
+      arch: process.arch,
+      cwd: process.cwd(),
+      argv: process.argv,
+      env: process.env,
```

Comment:
Ran this flag on Windows just now and it worked fine, but I'm hesitant about returning `env`. I understand that it may point out issues, but it can also contain tons of either useless or sensitive vars, and that's not including whatever is in the Path variable. It can also contain a lot of entries, which can make it annoying for the end user to filter out the sensitive info, should they choose to do so.
(also this showed that I don't have `APPIUM_HOME` set at all, yet UiAutomator2 still works without issues. 🤷 )

Reply:
This flag is created for debug usage, so it's ok to print the env. It won't be shared anywhere beside the local console, although is still useful to debug possible issues.

> (also this showed that I don't have APPIUM_HOME set at all, yet UiAutomator2 still works without issues. 🤷 )

it must not be set explicitly. The server can autodetect the value in most of the cases.